### PR TITLE
When archiving fails with a SQL exception, also output the plugin name in the exception message to help troubleshooting

### LINKED ISSUE #8600: When archiving fails with a SQL exception, also output the plugin name in the exception message to help troubleshooting
when running scheduled archiving, via the `core:archive` command, sometimes one of the sub-processes fails.

Here is an example error we got on production: `ERROR [2015-08-19 16:06:42] Got invalid response from API request: ?module=API&method=API.get&idSite=191124&period=day&date=last52&format=php&trigger=archivephp. Response was 'a:2:{s:6:"result";s:5:"error";s:7:"message";s:32:"Invalid bind-variable name ':00'";}'`

The challenge is that we do not know where this error was triggered. It could be in core, in a core plugin, or in any third party plugin used by this server.

Suggestion:

 * can we display in the error message, the plugin name that triggered this error?
  * this is similar to our recent improvement that displays the plugin name in the SQL query itself as a SQL comment: #7913

What do you think?

(adding to LTS since it will help troubleshooting very much!)

### PR DESCRIPTION
**Pull request by tsteur**
fixes #8600

It should append eg `- caused by plugin Actions`. Not sure if there's a better way but seems to do the job.

An example output:

```
INFO [2015-08-20 09:35:47] Will pre-process for website id = 1, period = day, date = last52
INFO [2015-08-20 09:35:47] - pre-processing all visits
ERROR [2015-08-20 09:35:49] Got invalid response from API request: ?module=API&method=API.get&idSite=1&period=day&date=last52&format=php&trigger=archivephp. Response was 'a:2:{s:6:"result";s:5:"error";s:7:"message";s:127:"SQLSTATE[HY093]: Invalid parameter number: number of bound variables does not match number of tokens - caused by plugin Actions";}'
```

### CODE REVIEW

File: core/ArchiveProcessor/PluginsArchiver.php [hunk 1]
```diff
@@ -114,25 +115,32 @@ public function callAggregateAllPlugins($visits, $visitsConverted)
...
-                $this->logAggregator->setQueryOriginHint('');
+                    Log::debug("PluginsArchiver::%s: %s while archiving %s reports for plugin '%s'.",
+                        __FUNCTION__,
+                        $timer->getMemoryLeak(),
+                        $this->params->getPeriod()->getLabel(),
+                        $pluginName
+                    );
+                } catch (Exception $e) {
+                    $Klass = get_class($e);
+                    $exception = new $Klass($e->getMessage() . " - caused by plugin $pluginName", $e->getCode(), $e);
```
Comment (diosmosis) [inline core/ArchiveProcessor/PluginsArchiver.php]: is `$Klass` a typo (ie, should it be `$klass`)?

Comment (tsteur) [inline core/ArchiveProcessor/PluginsArchiver.php]: Not sure why I wrote `$Klass` :) Think we used to do it like this in another project when it contained classname... will rename

**Comment by diosmosis** [comment]
Is this something that can be tested? It would be good if we didn't regress this behaviour when refactoring the Archiver for 3.0.

Other than the above comment, looks good to me.

**Comment by tsteur** [comment]
Added a test, will merge. Didn't want to write a test initially since it's maybe overdoing the testing part but might be helpful

**Pull request merged**